README.md

```diff
@@ -1,7 +1,334 @@
 # Coveo C# Platform SDK
-The offical Coveo C# SDK allows your software to communicate with the Coveo Cloud Platform public APIs and the Push API.
+[![NuGet version (Coveo.Connectors.Utilities.PlatformSdk)](https://img.shields.io/nuget/v/Coveo.Connectors.Utilities.PlatformSdk.svg?style=flat-square)](https://www.nuget.org/packages/Coveo.Connectors.Utilities.PlatformSdk/)
 
-The code of the SDK is not open-source. This repo is meant to provide feedback, bug reporting, etc.
+The official Coveo C# SDK allows your software to communicate with the Coveo Cloud Platform public APIs and the Push API.
+
+The package is .NET Standard 2.0 compatible and therefore works with .NET Core and .NET Framework
+
+The code of the SDK is not open-source. This repository is meant to provide feedback, bug reporting, etc.
+
+## Installation
+Install the public NuGet package [Coveo.Connectors.Utilities.PlatformSdk](https://www.nuget.org/packages/Coveo.Connectors.Utilities.PlatformSdk/) in your C# project using your favorite IDE or NuGet command line.
+
+The package is signed before being published to NuGet to avoid pulling a malicious package.
+We provide the `PDBs` to have a better debugging experience.
+Documentation is available for most of the public members and methods.
 
 ## How to use the SDK
-Coming soon!
+First, you need to instantiate the client to interact with the platform. Here is the minimum configuration you need to provide:
+
+** Each section below will redirect you to the privileges needed for the requests to work. For more information about API key privileges, visit [Privilege Reference](https://docs.coveo.com/en/1707/cloud-v2-administrators/privilege-reference).**
+
+```csharp
+string apiKey = "Your api key with required rights";
+string organizationId = "Your organization ID";
+ICoveoPlatformConfig config = new CoveoPlatformConfig(apiKey, organizationId);
+ICoveoPlatformClient client = new CoveoPlatformClient(config);
+```
+
+By default, the package targets the US production environment (platform.cloud.coveo.com). If you want to target the US HIPAA environment (platformhipaa.cloud.coveo.com) you can do it this way:
+
+```csharp
+string apiKey = "Your api key with required rights";
+string organizationId = "Your organization ID";
+ICoveoPlatformConfig config = new CoveoPlatformConfig(Constants.Endpoint.UsEast1.HIPAA_PUSH_API_URL,
+    Constants.PlatformEndpoint.UsEast1.HIPAA_PLATFORM_API_URL,
+    apiKey,
+    organizationId);
+ICoveoPlatformClient client = new CoveoPlatformClient(config);
+```
+
+## Using the SDK to push documents to a Push API source
+Methods to interact with the Push API are part of the `client.DocumentManager` object.
+### Prerequisites
+1. Ensure your API key has access to push documents inside a Push API source.
```

Comment:
If by "access" you mean "privilege", I'd write "privilege" instead.

Reply:
```suggestion
1. Ensure your API key has the required privilege to push documents inside a Push API source.
```